backend/src/main/java/com/redhat/cloud/notifications/routers/handlers/endpoint/EndpointResourceCommon.java

```diff
@@ -0,0 +1,226 @@
+                includeLinkedEventTypes(endpoint.get().getEventTypes(), endpointDTO);
+            }
+            return endpointDTO;
+        }
+    }
+
+    /**
+     * Gets the list of endpoints.
+     * @param sec the security context of the request.
+     * @param query the page related query elements.
+     * @param targetType the types of the endpoints to fetch.
+     * @param activeOnly should only the active endpoints be fetched?
+     * @param name filter endpoints by name.
+     * @param authorizedIds set of authorized integrations that we are allowed
+     *                      to fetch.
+     * @return a page containing the requested endpoints.
+     */
+    protected EndpointPage internalGetEndpoints(
+        final SecurityContext sec,
+        final Query query,
+        final List<String> targetType,
+        final Boolean activeOnly,
+        final String name,
+        final Set<UUID> authorizedIds,
+        final boolean includeLinkedEventTypes
+    ) {
+        String orgId = getOrgId(sec);
+
+        List<Endpoint> endpoints;
+        Long count;
+
+        Set<CompositeEndpointType> compositeType;
+
+        if (targetType != null && targetType.size() > 0) {
+            compositeType = targetType.stream().map(s -> {
+                try {
+                    return CompositeEndpointType.fromString(s);
+                } catch (IllegalArgumentException e) {
+                    throw new BadRequestException("Unknown endpoint type: [" + s + "]", e);
+                }
+            }).collect(Collectors.toSet());
+        } else {
+            compositeType = Set.of();
+        }
+
+        endpoints = endpointRepository.getEndpointsPerCompositeType(orgId, name, compositeType, activeOnly, query, authorizedIds);
+        count = endpointRepository.getEndpointsCountPerCompositeType(orgId, name, compositeType, activeOnly, authorizedIds);
+
+        final List<EndpointDTO> endpointDTOS = new ArrayList<>(endpoints.size());
+        for (Endpoint endpoint: endpoints) {
+            // Fetch the secrets from Sources.
+            this.secretUtils.loadSecretsForEndpoint(endpoint);
+
+            // Redact the secrets for the endpoint if the user does not have
+            // permission.
+            this.redactSecretsForEndpoint(sec, endpoint);
+
+            EndpointDTO endpointDTO = endpointMapper.toDTO(endpoint);
+            if (includeLinkedEventTypes) {
+                includeLinkedEventTypes(endpoint.getEventTypes(), endpointDTO);
+            }
+            endpointDTOS.add(endpointDTO);
+        }
+
+        return new EndpointPage(endpointDTOS, new HashMap<>(), new Meta(count));
+    }
+
+    /**
+     * Removes the secrets from the endpoint's properties when returning them
+     * to the client.
+     * @param endpoint the endpoint to redact the secrets from.
+     */
+    @Deprecated(forRemoval = true)
+    protected void redactSecretsForEndpoint(final SecurityContext securityContext, final Endpoint endpoint) {
```

Comment:
This method is still present in `EndpointResource`. Should it be removed from that class?